crates/sui-types/src/object.rs

```diff
@@ -482,6 +481,34 @@ pub enum Owner {
     },
     /// Object is immutable, and hence ownership doesn't matter.
     Immutable,
+    /// Object is sequenced via consensus. Ownership is managed by the configured authenticator.
+    ///
+    /// Note: wondering what happened to `V1`? `Shared` above was the V1 of consensus objects.
+    ConsensusV2 {
+        /// The version at which the object most recently became a consensus object.
+        /// This serves the same function as `initial_shared_version`, except it may change
+        /// if the object's Owner type changes.
+        start_version: SequenceNumber,
+        /// The authentication mode of the object
+        authenticator: Box<Authenticator>,
+    },
+}
+
+#[derive(
+    Eq, PartialEq, Debug, Clone, Copy, Deserialize, Serialize, Hash, JsonSchema, Ord, PartialOrd,
+)]
+#[cfg_attr(feature = "fuzzing", derive(proptest_derive::Arbitrary))]
+pub enum Authenticator {
+    /// The contained SuiAddress exclusively has all permissions: read, write, delete, transfer
+    SingleOwner(SuiAddress),
+}
```

Comment:
Whats the thesis behind having a nested authenticator vs adding more variants to the owner enum? Or maybe put another way, do we expect to see more types of authenticators in the future?

Reply:
Yes, we do expect to see more types of authenticators, and we shouldn't in the future have to change a million places for a new Owner type when all we want to modify is how the object is authenticated

Reply:
another argument is that we might eventually have `FastPathV2` which would also contain an `Authenticator`, so we'd get some re-use there. (Although not all authenticators will necessarily be supported by `FastPathV2`)